Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -370,7 +370,15 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return FullSequence.GetHashCode() + DigestionParams.Protease.GetHashCode();
+            unchecked
```

Comment:
why unchecked?

Reply:
Good catch on the null exception. I'd suggest this instead:
```
int digestionParamsHash = DigestionParams == null ? 0 : DigestionParams.Protease == null ? 0 : DigestionParams.Protease.GetHashCode();
return (FullSequence ?? "").GetHashCode() ^ digestionParamsHash;
```

Reply:
If you need to update the DigestionParams GetHashCode() method, you should go ahead and do that.